pkg/datapath/linux/ipsec/ipsec_linux.go

```diff
@@ -491,82 +504,96 @@ func loadIPSecKeys(r io.Reader) (int, uint8, error) {
 	scanner.Split(bufio.ScanLines)
 	for scanner.Scan() {
 		var oldSpi uint8
-		var authkey []byte
-		offset := 0
+		var aeadKey, authKey []byte
+		offsetBase := 0
 
 		ipSecKey := &ipSecKey{
 			ReqID: 1,
 		}
 
-		// Scanning IPsec keys formatted as follows,
-		//    auth-algo auth-key enc-algo enc-key
+		// Scanning IPsec keys with one of the following formats:
+		// 1. [spi] aead-algo aead-key icv-len
+		// 2. [spi] auth-algo auth-key enc-algo enc-key [IP]
 		s := strings.Split(scanner.Text(), " ")
-		if len(s) < 2 {
-			return 0, 0, fmt.Errorf("missing IPSec keys or invalid format")
+		if len(s) < 3 {
+			return 0, 0, fmt.Errorf("missing IPSec key or invalid format")
 		}
 
-		spiI, err := strconv.Atoi(s[0])
+		spiI, err := strconv.Atoi(s[offsetSPI])
 		if err != nil {
 			// If no version info is provided assume using key format without
 			// versioning and assign SPI.
 			spiI = 1
-			offset = -1
+			offsetBase = -1
 		}
 		if spiI > linux_defaults.IPsecMaxKeyVersion {
-			return 0, 0, fmt.Errorf("encryption Key space exhausted, id must be nonzero and less than %d. Attempted %q", linux_defaults.IPsecMaxKeyVersion+1, s[0])
+			return 0, 0, fmt.Errorf("encryption key space exhausted. ID must be nonzero and less than %d. Attempted %q", linux_defaults.IPsecMaxKeyVersion+1, s[offsetSPI])
 		}
 		if spiI == 0 {
-			return 0, 0, fmt.Errorf("zero is not a valid key to disable encryption use `--enable-ipsec=false`, id must be nonzero and less than %d. Attempted %q", linux_defaults.IPsecMaxKeyVersion+1, s[0])
+			return 0, 0, fmt.Errorf("zero is not a valid key ID. ID must be nonzero and less than %d. Attempted %q", linux_defaults.IPsecMaxKeyVersion+1, s[offsetSPI])
 		}
 		spi = uint8(spiI)
 
-		keyLen, authkey, err = decodeIPSecKey(s[2+offset])
-		if err != nil {
-			return 0, 0, fmt.Errorf("unable to decode authkey string %q", s[1+offset])
-		}
-		authname := s[1+offset]
+		if len(s) > 6+offsetBase {
```

Comment:
what's `6`?

Reply:
6 is the length in the case the IP address field is present in the secret, so the offset of the IP address field plus 1. I've replaced by the appropriate constants.

Reply:
6 is the maximum size of the secret if all possible fields are present. See formats above. I've replaced by the appropriate constants.